src/pudl/transform/vcerare.py

```diff
@@ -283,107 +289,232 @@ def out_vcerare__hourly_available_capacity_factor(
     }
     # Combine the data and perform a few last cleaning mechanisms
     # Sort the data by primary key columns to produce compact row groups
-    return (
+    df = (
         _combine_all_cap_fac_dfs(clean_dict)
         .pipe(_combine_cap_fac_with_fips_df, fips_df)
         .sort_values(by=["state", "county_or_lake_name", "datetime_utc"])
         .reset_index(drop=True)
     )
+    parquet_path = _get_parquet_path()
+    parquet_path.mkdir(exist_ok=True)
+    df.to_parquet(
+        parquet_path / f"vcerare_hourly_available_capacity_factor-{year}",
+        index=False,
+    )
 
 
-@asset_check(
-    asset=out_vcerare__hourly_available_capacity_factor,
-    blocking=True,
-    description="Check that output table is as expected.",
-    op_tags={"memory-use": "high"},
-)
-def check_hourly_available_cap_fac_table(asset_df: pd.DataFrame):  # noqa: C901
-    """Check that the final output table is as expected."""
+@asset
+def out_vcerare__hourly_available_capacity_factor(
+    raw_vcerare__lat_lon_fips: pd.DataFrame,
+    raw_vcerare__fixed_solar_pv_lat_upv: pd.DataFrame,
+    raw_vcerare__offshore_wind_power_140m: pd.DataFrame,
+    raw_vcerare__onshore_wind_power_100m: pd.DataFrame,
+):
```

Comment:
This only creates a set of single-year parquet outputs, rather than creating a single concatenated parquet output that conforms to the PyArrow schema defined for the table, so it seems like there's a missing concatenation step that's required to get the outputs into the same shape as all of our other outputs -- a single `table_name.parquet` file in the `PUDL_OUTPUT/parquet` directory, similar to the use of `pq.ParquetWriter` in `pudl.etl.epacems_assets`.

But instead of having the transform embedded within that function as an op graph, it seems like we can just iterate over the annual files and create pyarrow Tables out of them, and write them serially to the single output file with `pq.ParquetWriter` using the PyArrow schema defined for the table in the PUDL resource metadata.  Using the resource metadata defined PyArrow schema also stamps the table and column descriptions into the Parquet metadata directly.

Reply:
Does the lower memory footprint of running the asset checks with DuckDB depend on the inputs being split into separate parquet files? Or will it still be memory efficient if it operates on a single concatenated parquet file that has annual row-groups?  Ideally we would want to validate the thing that we're actually going to distribute.

Reply:
I ran the VCE RARE ETL locally and the asset checks were a few 100 MB each of memory use, so DuckDB seems to be doing great!